### ISSUE TITLE
Provide means of "picking" magic strings rather than having to know the syntax (for field values, dictionary items etc.)

### ISSUE THREAD

**Issue by FransdeJong** [umbraco/Umbraco.Forms.Issues#553]
When adding a workflow you can use a magic string for example for the recipient Email.
In our case most of the times a editor is working with these workflows and they have no idea about syntax and a field alias.

It would be great if we could have a button behind the input "or choose a field" that shows all form fields for that form and after picking adds the magic string to the input.

This saves us a lot of support and makes the workflows friendlier to work with for editors.

**Comment by nul800sebastiaan** [comment]
That is a wonderful (and huge) feature request! For now, it's not something we're planning on doing but it's certainly a very user-friendly thing we should look into in the future. 💯

**Comment by FransdeJong** [comment]
I don't think this will be a huge request if you disregard tracking.

What I'm thinking about is the following:
- Behind (or near) the input there is a button/dropdown
- If you click that you get a list of fieldnames in the current form
- When you click on a name the alias of that field is put in the form with curlybraces around it.

A bit like adding a form macro to a template.
If it works a little like the rest of the backoffice this shouldn't be that hard and most of the time is spend on UI.

This is a huge help for editors who keep forgetting how the magic strings work and call us for support.

**Comment by nul800sebastiaan** [comment]
Ha! Okay, maybe huge is overstating it a little bit, but there's quite a significant bit of work that needs to be done to accomplish this 😄 We'd love to! But can't fit it in right now.

**Comment by ronaldbarendse** [comment]
This is related to https://github.com/umbraco/Umbraco.Forms.Issues/issues/329 and could be implemented using an external package that allows selecting pre-defined tags (fields, dictionary items, form name, URL, etc.) within text inputs: https://yaireo.github.io/tagify/#section-mix.

Like @nul800sebastiaan mentioned, this is quite a significant bit of work though and could be implemented as a new Forms setting field type within a package.

**Comment by FransdeJong** [comment]
@ronaldbarendse 
I don't think that is what I meant or I'm totally misunderstanding what you mean.
I also don't think I can add a setting to the existing workflows that Forms ships with?

So what I meant is add a small button behind the fields in the workflows that ship with forms (for example: "Send email with template (Razor) email fields"). If you click on that a list appears with all the formfields in the model, if you click on one the alias is placed in the input between curly braces. No validation, no tracking just a way to find the alias.

**Comment by FransdeJong** [comment]
@Amalie-Wowern 
Isn't it easier to reopen this one?

I'm happy to copy paste the request in a new issue but nothing will change and the comments will be lost.

**Comment by Amalie-Wowern** [comment]
Hi @FransdeJong 

I will reopen the issue.
